hazelcast/include/hazelcast/client/impl/metrics/metric_descriptor.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+
+#include <boost/optional.hpp>
+
+#include "hazelcast/util/export.h"
+
+namespace hazelcast {
+namespace client {
+namespace impl {
+namespace metrics {
+
+enum struct probe_unit
+{
+    BYTES = 0,
+    MS = 1,
+    NS = 2,
+    PERCENT = 3,
+    COUNT = 4,
+    BOOLEAN = 5,
+    ENUM = 6,
+
+    NONE = -1
+};
+
+class HAZELCAST_API metric_descriptor
```

Comment:
why not make this a simple struct with all public fields?

Reply:
why not do the same for all classes? :)

Reply:
This class does not have any method with some logic in it, hence, i think it is better for such cases to use struct. Some older classes reflected Java but I prefer this lately if no method with specific logic in it.

Reply:
I think there is "some" logic in this class. By making it a class and specifying defining ctors we enforce the ways how this type can be initialized. E.g.: if a descriptor name is given its value should also be given (it can't be null), the ctors are ensuring that.